### DISCUSSION TITLE
Using a variable to filter results from an XML query

### DISCUSSION THREAD

**Discussion by sridharvdp** [grafana/grafana-infinity-datasource#507] [Q&A]
Hi, 
Thank you for the amazing plugin.
I am new to Grafana and I am trying to apply a "filter" based on a variable. 
The file type I am using is XML and I use the backend parser to translate the XML.
In the example I show in the picture below, when I put in ID=="Full_String" in the field Filter, it works but when I try to put a variable eg: (ID==$variable) that contains this string, it throws an error Invalid Token :'$' 
@yesoreyeram Can you please let me know if we are allowed to declare variables here. If yes, can you help me with the good syntax to apply a filter with a variable that contains a string please  ?

![Snip](https://user-images.githubusercontent.com/60777628/212677591-e87647d4-7cd2-4ae2-9a3c-a2f227c27b41.jpg)

Thank you!

Sridhar

**Comment by yesoreyeram** [comment]
I guess your filter syntax is incorrect. Probably you need to wrap you var name with single quote. Example `ID == '$variable1'`. Give a try.

**Reply by yesoreyeram** [reply]
@sridharvdp did it help?

**Reply by sridharvdp** [reply]
@yesoreyeram Thank you for your quick reply. No, It does not work. The filter seems to look for the exact string declared inside single quotes and not consider it as a variable. I made a quick test to check this. Therefore, it does not return any data. Do you have other ideas for the syntax ? Do you think this is an enhancement ?
![snip2](https://user-images.githubusercontent.com/60777628/213140879-29995ca2-a46f-4112-9add-88f10fa99aa7.JPG)

**Reply by yesoreyeram** [reply]
I doubt this is some issue with your variable setup.

<img width="1148" alt="image" src="https://user-images.githubusercontent.com/153843/213154666-7be60d65-6fd5-4bd1-ba2d-9dfdc5c7a4d9.png">

Can you post a Sample XML and your dashboard json for further debugging?

**Reply by sridharvdp** [reply]
Strange to see that it works in your example when you use CSV and not with XML. Here is the sample XML and dashboard with the variable configured for your test. Let me know your finding.

[Sample Data+Dashboard.zip](https://github.com/yesoreyeram/grafana-infinity-datasource/files/10447613/Sample.Data%2BDashboard.zip)

**Reply by yesoreyeram** [reply]
Ah.. Found an issue. `xml` and `html` were missing in the variable interpolation step.

https://github.com/yesoreyeram/grafana-infinity-datasource/blob/4dff0d3236c8798db6e8edadc8338e876471fb6d/src/interpolate.ts#L66

Fix in the next release. Thank you for reporting.
